### PR TITLE
OnDeviceStatusChanged

### PR DESCRIPTION
**Pull request by forwardpointer**
### Type

- [ ] Bug fix
- [x] Feature addition
- [ ] Feature update
- [ ] Breaking change
- [ ] Non-functional change
- [ ] Documentation
- [ ] Infrastructure

### Goals

* Implement the handler function OnDeviceStatusChanged
* Implement the conversion to neutral type for deviceStatus, genericError, sessionStatus
 
### Technical Details

* Added an Uninitialized value to the enum UwbDeviceState so we can print something different when the device is first initialized.
* Added some static const maps from ddi type to neutral type

### Test Results
None yet

### Checklist

- [x] Build target `all` compiles cleanly.
- [x] clang-format and clang-tidy deltas produced no new output.
- [x] Newly added functions include doxygen-style comment block.

### CODE REVIEW

File: lib/uwb/UwbDevice.cxx [hunk 1]
```diff
@@ -41,7 +41,15 @@ UwbDevice::OnStatusChanged([[maybe_unused]] UwbStatus status)
...
 UwbDevice::OnDeviceStatusChanged([[maybe_unused]] UwbStatusDevice statusDevice)
 {
-    // TODO: implement this
+    if (m_status.State == UwbDeviceState::Uninitialized) {
+        m_status = statusDevice;
+        PLOG_VERBOSE << "initialized device status as " << m_status.ToString();
+        return;
+    }
+    if (m_status.State != statusDevice.State) {
+        PLOG_VERBOSE << "changed state: " << m_status.ToString() << " --> " << statusDevice.ToString();
```
Comment (abeltrano) [inline lib/uwb/UwbDevice.cxx:50]: nit: We have to `ToString()` helper such that you can do this instead without invoking `ToString()` directly:

```
#include <notstd/tostring.hxx>
using namespace strings::ostream_operators
```
...
```suggestion
    if (m_status.State == UwbDeviceState::Uninitialized) {
        m_status = statusDevice;
        PLOG_VERBOSE << "initialized device status as " << m_status;
        return;
    }
    if (m_status.State != statusDevice.State) {
        PLOG_VERBOSE << "changed state: " << m_status << " --> " << statusDevice;
```

Comment (forwardpointer) [reply lib/uwb/UwbDevice.cxx:50]: I tried your suggestion but it complained it couldn't find the operator<< for m_status :(

File: windows/devices/uwb/UwbCxAdapterDdiLrp.cxx [hunk 2]
```diff
@@ -534,10 +534,95 @@ windows::devices::uwb::ddi::lrp::To(const UWB_DEVICE_CAPABILITIES &deviceCapabil
...
+    }
+    case UWB_NOTIFICATION_TYPE_GENERIC_ERROR: {
+        auto enumId = notstd::to_underlying(notificationData.genericError);
+        if (enumId < notstd::to_underlying(UWB_STATUS_ERROR_SESSION_NOT_EXIST)) {
+            return StatusToMapGeneric.at(notificationData.genericError);
+        }
+        if (enumId < notstd::to_underlying(UWB_STATUS_RANGING_TX_FAILED)) {
+            return StatusToMapSession.at(notificationData.genericError);
+        }
+        return StatusToMapRanging.at(notificationData.genericError);
```
Comment (abeltrano) [inline windows/devices/uwb/UwbCxAdapterDdiLrp.cxx:613]: We don't need a special case for this. We should just pass `notificationData.genericError` to the `::To(UWB_STATUS& status)` function which returns a `UwbStatus` (I suggested writing this function in another comment).

**Review by abeltrano — CHANGES_REQUESTED** [review]

File: lib/uwb/UwbDevice.cxx (hunk 1 above)
Comment (abeltrano) [reply lib/uwb/UwbDevice.cxx:50]: Hmm, probably the `PLOG` macro isn't actually an `ostream`. Oh well :'(

**Review by abeltrano — APPROVED** [review]
This is great.

**Pull request merged**